### PR TITLE
Replace lazy hashset with matches!

### PR DESCRIPTION
**Pull request by untitaker**
llvm will typically create much faster code than accessing a hashset at
runtime

~source: trust me bro~ actually ran benchmarks now

### CODE REVIEW

**Comment by untitaker** [comment]
ok

```
extract from large docs time:   [52.441 ms 53.355 ms 55.164 ms]
                        change: [-27.284% -23.363% -19.249%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 2 outliers among 10 measurements (20.00%)
  1 (10.00%) high mild
  1 (10.00%) high severe
```

File: lychee-lib/src/extract/mod.rs [hunk 1]
```diff
@@ -8,26 +6,17 @@ mod markdown;
...
-        "samp".into(),
-        "xmp".into(),
-        "plaintext".into(),
-        "listing".into(),
-    ])
-});
-
 /// Check if the given element is in the list of preformatted tags
+///
+/// preformatted tags are HTML elements that are deemed verbatim (i.e. preformatted).  These will
```
Comment (mre) [inline lychee-lib/src/extract/mod.rs:13]: The wording is a little weird. Reads a bit tautological: "preformatted tags are preformatted". But I get your point. 😅

**Review by mre — APPROVED** [review]

**Comment by mre** [comment]
Wait, am I reading this correctly? This gives us a 20% perf improvement on extracting tags!? That's insane.

**Comment by untitaker** [comment]
yeah. it's a bit of guesswork. there's other hashsets in the code where i expected a similar perf gain, (e.g. example_domains), but i'm unable to replicate any sort of performance improvement by replacing those

File: lychee-lib/src/extract/mod.rs (hunk 1 above)
Comment (untitaker) [reply lychee-lib/src/extract/mod.rs]: yeah that's a copypaste error. didn't read it well enough

**Comment by mre** [comment]
It hadn't crossed my mind to use `matches` actually. If not for anything else it will at least reduce the code complexity a bit, so if you want, feel free to change the other occurrences as well. But we can do that in another pull request.

**Pull request merged**